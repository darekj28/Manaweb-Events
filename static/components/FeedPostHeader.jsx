var React = require('react');
export default class FeedPostHeader extends React.Component {
	constructor(props) {
		super(props);
		this.handleFilterUser = this.handleFilterUser.bind(this);
	}
	handleFilterUser() {
		this.props.handleFilterUser(this.props.userID);
	}
	render() {
		var tpc = "";
		if (this.props.post.isTrade) tpc = tpc.concat(' Trade ');
		if (this.props.post.isPlay)  tpc = tpc.concat(' Play ');
		if (this.props.post.isChill) tpc = tpc.concat(' Chill ');
		return(
			<div className="FeedPostHeader">
				<div className="tpc pull-right"><kbd>{tpc}</kbd></div> 
				<div className="headerpart name" onClick={this.handleFilterUser}><b>{this.props.post.name}</b></div>
				<div className="headerpart username text-muted" onClick={this.handleFilterUser}>@{this.props.post.userID}</div>
				<div className="headerpart time text-muted">&#8226; {this.props.post.time}</div>
				{ (this.props.isAdmin || !this.props.isOP) && 
						<div className="dropdown pull-left" id={"dropdown_" + this.props.post.comment_id} onClick={this.props.scrollToDropdown}>
							<a href="#" className="dropdown-toggle" data-toggle="dropdown">
				                <span className="glyphicon glyphicon-option-horizontal 
				                				pull-left AppGlyphicon"></span>
				            </a>
				            <ul className="PostDropdown pull-left dropdown-menu">
				              	{(this.props.isAdmin) && <li><a id="hpe" href="#" onClick={this.props.handlePostEdit}>Edit post</a></li> }
			              		{(this.props.isAdmin) && <li><a id="hpd" href="#" onClick={this.props.handlePostDelete}>Delete post</a></li> }
			              		{(!this.props.isOP || this.props.isAdmin) && <li><a id="hpr" href="#" onClick={this.props.handlePostReport}>Report post</a></li> }
				            </ul>
				        
				        </div>
				  	  }	
			</div>
			)
	}
}