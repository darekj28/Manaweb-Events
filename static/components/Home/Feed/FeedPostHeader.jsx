var React = require('react');
export default class FeedPostHeader extends React.Component {
	handleFilterUser() {
		this.props.handleFilterUser(this.props.post.userID);
	}
	render() {
		return(
			<div className="FeedPostHeader">
				{this.props.post.isChill && 
				<div className="postType pull-right">
					<span className="HeaderGlyphicon fa fa-snowflake-o"></span>
				</div>}
				{this.props.post.isTrade && <div className="postType pull-right">
					<span className="HeaderGlyphicon fa fa-handshake-o"></span>
				</div> }
				{this.props.post.isPlay && 
				<div className="postType pull-right">
					<span className="HeaderGlyphicon fa fa-play"></span>
				</div> } 				
				<div className="headerpart name" onClick={this.handleFilterUser.bind(this)}><b>{this.props.post.name}</b></div>
				{this.props.post.userID != "$DELETED_USER" && <div className="headerpart username text-muted">@{this.props.post.userID}</div>}
				<div className="headerpart time text-muted">&#8226; {this.props.post.time}</div>
			</div>
			)
	}
}