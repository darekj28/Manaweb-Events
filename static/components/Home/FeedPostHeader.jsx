var React = require('react');
export default class FeedPostHeader extends React.Component {
	constructor(props) {
		super(props);
		this.handleFilterUser = this.handleFilterUser.bind(this);
	}
	handleFilterUser() {
		this.props.handleFilterUser(this.props.post.userID);
	}
	render() {
		var postType = "";
		if (this.props.post.isTrade) postType = postType.concat(' trade ');
		if (this.props.post.isPlay)  postType = postType.concat(' play ');
		if (this.props.post.isChill) postType = postType.concat(' chill ');
		return(
			<div className="FeedPostHeader">
				<div className="postType pull-right">{postType}</div> 
				<div className="headerpart name" onClick={this.handleFilterUser}><b>{this.props.post.name}</b></div>
				<div className="headerpart username text-muted">@{this.props.post.userID}</div>
				<div className="headerpart time text-muted">&#8226; {this.props.post.time}</div>
			</div>
			)
	}
}