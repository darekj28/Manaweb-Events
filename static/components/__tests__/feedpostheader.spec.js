import FeedPostHeader from '../FeedPostHeader.jsx';

describe('FeedPostHeader', () => {
	it('renders correctly', () => {
		const fph = renderer.create(
			<FeedPostHeader name="a" userID="a" handleFilterUser={jest.fn()} 
				isTrade={true} isPlay={true} isChill={true} time="a"/>
			);
		expect(fph).toMatchSnapshot();
	});
});