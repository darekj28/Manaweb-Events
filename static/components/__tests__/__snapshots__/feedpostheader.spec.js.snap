exports[`FeedPostHeader renders correctly 1`] = `
<div
  className="FeedPostHeader">
  <div
    className="tpc pull-right">
    <kbd>
       Trade  Play  Chill 
    </kbd>
  </div>
  <div
    className="headerpart name"
    onClick={[Function]}>
    <b>
      a
    </b>
  </div>
  <div
    className="headerpart username text-muted"
    onClick={[Function]}>
    @
    a
  </div>
  <div
    className="headerpart time text-muted">
    â€¢ 
    a
  </div>
</div>
`;
